doc/pages/options.asciidoc

```diff
@@ -139,20 +139,27 @@ are exclusively available to built-in options.
     `set -remove` removes the given specs from the list. +
 
 *completions*::
-    a list of `<text>|<select cmd>|<menu text>` candidates,
+    a list of `<text>[|<filter text>]|<select cmd>|<menu text>` candidates,
     except for the first element which follows the
     `<line>.<column>[+<length>]@<timestamp>` format to define where the
     completion apply in the buffer.
     Any `|` or `\` characters that occur within the `<text>`,
-    `<select cmd>`, or `<menu text>` fields should be escaped as `\|`
-    or `\\`.
-
-    Select commands are arbitrary Kakoune commands that will be executed
-    each time the element is selected in the menu. The common use case is
-    to display element specific documentation.
-
-    Markup can be used in the menu text.
-    (see <<faces#markup-strings,`:doc faces markup-strings`>>)
+    `<filter text>`, `<select cmd>`, or `<menu text>` fields should be
+    escaped as `\|` or `\\`.
+
+    Options of this type are are meant to be added to the `completers`
+    option to provide insert mode completion.  Candidates are shown if the
+    text typed by the user (between `<line>.<colum>` and the cursor) is a
+    prefix of `<filter text>` (if provided) or `<text>`.
+
+    For each remaining candidate, the completion menu displays `<filter text>`
+    (if provided) or `<text>`, followed by `<menu text>`, which can contain
+    Markup (see <<faces#markup-strings,`:doc faces markup-strings`>>).
+
+    If the user selects an item from the completion menu, the text they typed
```

Comment:
```suggestion
    As the user selects items from the completion menu, the text they typed
```

With the original wording, it sounded like these actions occurred once the user had accepted a completion, rather than just while they cycle through the completione.